News topics

felixbiessmann opened the pull request:
hab mal nen topic modelling fuer die news gemacht;

was ich noch machen will:
- flask api rum
- api queried manifesto_model fuer pol. einordnung (periodisch, wenn get_news gemacht wird)
- news koennen abgefragt werden fuer 'left' oder 'right' von anderen micro-services; 

damit koennte man dann in rightornot oder anderen apps aktuelle news je nach politischer einordnung bekommen.

Commit:
topic modelling for newsreader

Commit:
added stopwords

tdhd commented:
Cool! Möchtest du den Crawler auch hinter ne Flask API packen?

tdhd commented:
Wir sollten am besten die news in einer separaten DB halten, also mongo, oder damits vielleicht einheitlicher ist auch sqlite3. Was meinst du?

Commit:
news reader api draft

Commit:
fixed some bugs

Commit:
Added news reader functionality to rightornot quiz
 - after completion of active learning part, some news from opposite political spectrum are shown

tdhd commented on services/docker-compose.yml:
```diff
@@ -1,44 +1,46 @@
 version: '3'
 services:
-  persistence:
```
gibt es den container nicht mehr?

tdhd commented on services/news_crawler/newsreader.py:
```diff
@@ -1,21 +1,51 @@
+        self.n_topics = n_topics
+
+    @staticmethod
+    def get_topics(texts, n_topics=5):
```
sehr cool!

tdhd commented on services/persistence/Dockerfile:
```diff
@@ -1,15 +0,0 @@
-#FROM ubuntu:16.04
```
ist die Idee, dass wir halt einfach die sqlite db in die entsprechenden container einhaengen? dann brauchen wir ja wirklich keinen persistence container mehr.

tdhd commented on services/user_interface/main.py:
```diff
@@ -127,6 +128,62 @@ def user_labels():
+    """
+    n_topics = int(request.args.get('n', default=5))
+    r = requests.get(
+        'http://news_crawler:{}/get_topics?n={}'.format(
```
gibt es diesen endpoint?

tdhd commented on services/user_interface/main.py:
```diff
@@ -127,6 +128,62 @@ def user_labels():
+@app.route('/get_news')
+def get_news():
+    r = requests.get(
+        'http://news_crawler:{}/get_news?bias={}'.format(
```
und gibt es auch den endpoint beim news crawler?

felixbiessmann replied:
sorry, hatte vergessen einzuchecken, ja gibts und hat auch funktioniert. sieht noch scheisse aus, und die linksrechts einordnung is auch noch nich so super, kommt viel muell. aber geht erstma

tdhd reviewed (commented):
toll, dass du das setup vereinfacht hast, also den persistence container rausgeworfen hast.

Commit:
forgot to add api.py

Commit:
- minor layout stuff
- per topic selection of news (not used)
- restriction of number of news

Commit:
plotting for paper

tdhd commented:
Ist das der gleiche Conny wie in dem black and white plot PR?

felixbiessmann commented:
conny?

> On 14. Jan 2018, at 20:36, Philipp Schmidt <notifications@github.com> wrote:
> 
> Ist das der gleiche Conny wie in dem black and white plot PR?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

tdhd commented:
Ah sorry, commit

felixbiessmann commented:
Ja, war's, hatte vergessen in deinen branch zu wechseln vorher, sorry.

Commit:
einfacheres layout

Commit:
Merge branch 'master' into news_topics
